config/src/test/kotlin/org/springframework/security/config/web/servlet/AuthorizeRequestsDslTests.kt

```diff
@@ -271,4 +274,92 @@ class AuthorizeRequestsDslTests {
             }
         }
     }
+
+    @EnableWebSecurity
+    @EnableWebMvc
+    open class AuthorizeRequestsByMvcConfigWithHttpMethod : WebSecurityConfigurerAdapter() {
+        override fun configure(http: HttpSecurity) {
+            http {
+                authorizeRequests {
+                    authorize(HttpMethod.GET, "/path", permitAll)
+                    authorize(HttpMethod.PUT, "/path", denyAll)
+                }
+            }
+        }
+
+        @RestController
+        internal class PathController {
+            @GetMapping("/path")
+            fun permit() {
+            }
+
+            @PutMapping("/path")
+            fun deny() {
+            }
+        }
+    }
+
+    @Test
+    fun `request when secured by mvc with http method then responds`() {
+        this.spring.register(AuthorizeRequestsByMvcConfigWithHttpMethod::class.java).autowire()
+
+        this.mockMvc.get("/path")
+            .andExpect {
+                status { isOk }
+            }
+
+        this.mockMvc.put("/path")
+            .andExpect {
+                status { isForbidden }
+            }
+    }
+
+    @EnableWebSecurity
+    @EnableWebMvc
+    open class AuthorizeRequestsByMvcConfigWithAndWithoutHttpMethod : WebSecurityConfigurerAdapter() {
```

Comment:
I had a different test in mind when I mentioned using HTTP method and servlet path.

We should have a test similar to `` `request when secured by mvc with servlet path then responds based on servlet path` ``, but using the HTTP method.

The configuration would look something like
```
http {
    authorizeRequests {
        authorize(HttpMethod.GET, "/path", "/spring", denyAll)
    }
}
```

Reply:
Sorry I misunderstood. I'll look at replacing the second test this weekend.

Reply:
Absolutely, good catch.